src/CommunityToolkit/Xamarin.CommunityToolkit/AttachedProperties/SetFocusOnEntryCompleted.shared.cs

```diff
@@ -0,0 +1,47 @@
+﻿using Xamarin.Forms;
+
+namespace Xamarin.CommunityToolkit.AttachedProperties
+{
+	/// <summary>
+	/// The <see cref="SetFocusOnEntryCompleted"/> is an attached property for entries that allows the user to specify what <see cref="VisualElement"/> should gain focus after the user completes that entry.
+	/// </summary>
+	public class SetFocusOnEntryCompleted
+	{
+		/// <summary>
+		/// The <see cref="NextElementProperty"/> attached property.
+		/// </summary>
+		public static readonly BindableProperty NextElementProperty =
+			BindableProperty.CreateAttached(
+				"NextElement",
+				typeof(VisualElement),
+				typeof(SetFocusOnEntryCompleted),
+				default(VisualElement),
+				propertyChanged: OnNextElementChanged);
+
+		/// <summary>
+		/// Required <see cref="GetNextElement"/> accessor for <see cref="NextElementProperty"/> attached property.
+		/// </summary>
+		public static VisualElement GetNextElement(BindableObject view)
+		{
+			return (VisualElement)view.GetValue(NextElementProperty);
+		}
+
+		/// <summary>
+		/// Required <see cref="SetNextElement"/> accessor for <see cref="NextElementProperty"/> attached property.
+		/// </summary>
+		public static void SetNextElement(BindableObject view, VisualElement value)
+		{
+			view.SetValue(NextElementProperty, value);
+		}
+
+		static void OnNextElementChanged(BindableObject bindable, object oldValue, object newValue)
+		{
+            var entry = bindable as Entry;
+
+            if (entry == null)
+                return;
+
+            entry.Completed += (s, e) => GetNextElement(entry)?.Focus();
```

Comment:
This can be dangerous since we don't unsubscribe from this event this can lead to memory leaks.

Reply:
Hmm, I don't instantly know how to improve this; I'll look into weak event pattern stuff.

Reply:
That may not help here, because we aren't implementing the event. Maybe [this can help you](https://github.com/xamarin/XamarinCommunityToolkit/blob/0d55a6340508de0bd63be4c44d484e0823529d5c/src/CommunityToolkit/Xamarin.CommunityToolkit/ObjectModel/Extensions/INotifyPropertyChangedEx.shared.cs#L8)

Reply:
That helps a lot, thanks; will use a WeakReference<T> to the Entry.